app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -151,9 +151,9 @@ def fetch_availability
           metric_id = @ems.build_metric_id('A', deployment, 'Deployment Status~Deployment Status')
           metric_ids[metric_id] = deployment
         end
-        availabilities = @ems.metrics_client.avail.raw_data(metric_ids.keys, :limit => 1, :order => 'DESC')
-        availabilities.each do |availability|
-          metric_ids[availability['id']][:status] = process_availability(availability['data'].first)
+        unless metric_ids.empty?
+          availabilities = @ems.metrics_client.avail.raw_data(metric_ids.keys, :limit => 1, :order => 'DESC')
+          parse_availability availabilities, metric_ids
```

Comment:
I know, it's not part of your PR, but can we change the name of `metric_ids` variable? `s/metric_ids/resources_by_metric_id/`